ocs/agent/aggregator.py

```diff
@@ -149,10 +150,79 @@ def empty(self):
 
         return True
 
-    def write(self, data):
+    def _verify_provider_data(self, data):
+        """Check the provider data for invalid field names. Meant to be used in
+        combination with Provider._rebuild_invalid_data().
+
+        Args:
+            data (dict): data dictionary passed to Provider.save_to_block()
+
+        Returns:
+            bool: True if all field names valid. False if any invalid names found
+
+        """
+        verified = True
+        for block_name, block_dict in data.items():
+            for field_name, field_values in block_dict['data'].items():
+                try:
+                    ocs_feed.Feed.verify_data_field_string(field_name)
+                except ValueError:
+                    self.log.error("data field name {field} is " +
+                                   "invalid, removing invalid characters.",
+                                   field=field_name)
+                    verified = False
+
+        return verified
+
+    def _rebuild_invalid_data(self, data):
+        """Rebuild an invalid data dictionary.
+
+        Args:
+            data (dict): data dictionary passed to Provider.save_to_block().
+
+        Returns:
+            dict: A rebuilt data dictionary with invalid characters stripped
+                  from the field names, limited to 255 characters in length.
+
         """
-        Saves a list of data points into blocks.
-        A block will be created for any new block_name.
+        new_data = {}
+        for block_name, block_dict in data.items():
+            new_data[block_name] = {}
+            # rebuild block_dict
+            for k, v in block_dict.items():
+                if k == 'data':
+                    new_data[block_name]['data'] = {}
+                    for field_name, field_values in block_dict['data'].items():
+                        # replace invalid characters, and limit to 255 characters
+                        new_field_name = re.sub('[^a-zA-Z0-9_]', '', field_name)[:255]
```

Comment:
I think this replacement should also remove invalid leading characters.

Reply:
It will remove those as written:
```
>>> import re
>>> re.sub('[^a-zA-Z0-9_]', '', "!@#invalid.leading.characters")
'invalidleadingcharacters'
```

But maybe I don't follow what you're saying?

Reply:
```
>>> import re
>>> re.sub('^a-zA-Z0-9_]', '', '1234invalid.leading')
'1234invalid.leading'
```

Reply:
Ah, good catch! Will fix.

Reply:
Alright, addressed in f566f18. Maybe some regex magic could solve it easier, but I broke down each step into somewhat easy to read parts.